### PR TITLE
Fixes #30602 - Use new react mounter for form_helper

### PR DESCRIPTION
**Pull request by yifatmakias**

### CODE REVIEW

File: app/helpers/form_helper.rb [hunk 1]
```diff
@@ -269,7 +269,7 @@ def submit_or_cancel(f, overwrite = false, args = { })
 
   def react_cancel_button(args)
     content_tag(:span, :id => 'form-cancel-btn') do
-      mount_react_component('RedirectCancelButton', '#form-cancel-btn', { :cancelPath => args[:cancel_path] }.to_json, :flatten_data => true)
+      react_component('RedirectCancelButton', { :cancelPath => args[:cancel_path] })
```
Comment (Ron-Lavi) [inline app/helpers/form_helper.rb]: Looks better :)
can you also remove the `content_tag` on line 271?

**Comment by yifatmakias** [comment]
[test katello]

**Review by amirfefer — DISMISSED** [review]
Thanks @yifatmakias !

It looks like a regression - the cancel button is not mounted at all:
no-cancel-btn

**Comment by yifatmakias** [comment]
@amirfefer weirdly the cancel button did not appear before the change, you can see in developed. I will check this.

**Comment by yifatmakias** [comment]
[test katello]

**Comment by yifatmakias** [comment]
[test katello]

**Comment by xprazak2** [comment]
Something strange is happening with the button, it was not showing for me either. The strange thing is that the mounting js was not executed at all. I started poking the helper method and did the following in form_helper:

```rb
def react_cancel_button(args)
   btn = mount_react_component('RedirectCancelButton', '#form-cancel-btn', { :cancelPath => args[:cancel_path] }.to_json, :flatten_data => true)
    content_tag(:span, btn, :id => 'form-cancel-btn')
  end
```

But the most curious is that when I reverted back, the button was suddenly there as well!

**Comment by yifatmakias** [comment]
@xprazak2 What do you mean by reverted back? from what did you reverted back?

**Comment by yifatmakias** [comment]
[test katello]

**Comment by yifatmakias** [comment]
[test katello]

**Comment by xprazak2** [comment]
I think I have figured it out: https://github.com/theforeman/foreman/pull/7957

**Comment by yifatmakias** [comment]
This PR is blocked by another PR:
https://github.com/theforeman/foreman/pull/7957

**Comment by yifatmakias** [comment]
[test foreman]

**Comment by yifatmakias** [comment]
[test foreman]

**Comment by xprazak2** [comment]
No longer blocked by #7957.

**Comment by yifatmakias** [comment]
[test foreman]

**Review by ezr-ondrej — APPROVED** [review]
Thanks @yifatmakias ! :+1: 
Thanks everyone involved in fixing the missing button (mainly @xprazak2 and @MariaAga).
And thanks @xprazak2 and @amirfefer for reviews! ^_^

**Pull request merged**